### PR TITLE
fix: cover dense temp-file and sparse k-mer edge cases

### PR DESCRIPTION
**Pull request by cjprybol**
## Summary
- fix the dense temp-file path so the pass-2 progress lock always exists
- guard empty multi-scale consensus selection to avoid slicing past an empty k-mer list
- add focused coverage tests for sparse result saving, missing rarefaction data, dense temp-file error recovery, UInt32 promotion, and forced dense DNA counting

## Verification
- ran a focused Julia assertion probe covering the malformed dense-input path, sparse result-file save path, and empty-consensus multi-scale case
- reran the same focused probe with `--code-coverage=user`, producing `src/kmer-analysis.jl.66245.cov` with hits on the touched branches

## Review
- manual review lenses: correctness, quality, and security found no additional issues in the final diff
- `coderabbit review --plain --type committed --base master` could not run because the account lacks the required Hoppy CLI addon

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved robustness of k-mer analysis result handling and metadata preservation.
  * Fixed edge-case handling in consensus k-mer selection to prevent out-of-bounds errors.

* **Tests**
  * Significantly expanded test coverage for k-mer analysis workflows, including persistence, frequency analysis, and edge cases.
  * Added comprehensive test suite for related utilities and helper functions across multiple alphabet types.

### CODE REVIEW

File: src/kmer-analysis.jl [hunk 1]
```diff
@@ -89,12 +89,13 @@ function load_kmer_results(filename::AbstractString)
...
-            try
-                metadata["alphabet"] = Symbol(loaded_data["metadata/alphabet"])
-            catch
-                @warn "Could not convert loaded alphabet '$(loaded_data["metadata/alphabet"])' back to Symbol. Storing as String."
-                metadata["alphabet"] = loaded_data["metadata/alphabet"]
+            loaded_alphabet = loaded_data["metadata/alphabet"]
+            if loaded_alphabet isa Symbol
+                metadata["alphabet"] = loaded_alphabet
+            elseif loaded_alphabet isa AbstractString
+                metadata["alphabet"] = Symbol(loaded_alphabet)
```
[the remark this answers is not in the document]
Comment (cjprybol) [reply src/kmer-analysis.jl]: Addressed in c1310f20. `load_kmer_results` now keeps the string-to-`Symbol` conversion guarded locally, warns, and preserves the original `metadata/alphabet` value if conversion fails, so malformed persisted strings no longer cause the whole load to return `nothing`. Added a regression case with a NUL-containing alphabet string in `test/3_feature_extraction_kmer/kmer_analysis_coverage_expansion_test.jl`, and verified with `julia --project=. test/3_feature_extraction_kmer/kmer_analysis_coverage_expansion_test.jl` (60/60 passing).

**Comment by cjprybol** [comment]
Closing as superseded by #293, merged as `a261c828028e87a291f4561ba86895979adb76e5`.

#293 carries this dense temp-file / sparse k-mer edge-case work forward, rebased on current `master`, with review repairs for malformed alphabet metadata fallback and `force_threading=false` in dense pass 2. It was validated with:

- `julia --project=. -e 'include("test/3_feature_extraction_kmer/kmer_analysis_additional_coverage_test.jl")'`
- `julia --project=. -e 'include("test/3_feature_extraction_kmer/kmer_analysis.jl"); include("test/3_feature_extraction_kmer/kmer_analysis_helpers_test.jl"); include("test/3_feature_extraction_kmer/kmer_analysis_coverage_expansion_test.jl"); include("test/3_feature_extraction_kmer/kmer_analysis_additional_coverage_test.jl")'`
